react-app/src/components/Extract.js

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { LinkContainer } from 'react-router-bootstrap'
+import { Col, Form, Row } from 'react-bootstrap'
+
+import '../stylesheets/Page.css';
+
+class Extract extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            configPath: "",
+            logPath: "",
+            includeDebug: false,
+            filterStart: false,
+            filterEnd: false
+        }
+
+        this.onConfigFileChange = this.onConfigFileChange.bind;
+        this.onLogFileChange = this.onLogFileChange.bind;
+        this.onIncludeDebugChange = this.onIncludeDebugChange.bind;
+        this.onFilterStartChange = this.onFilterStartChange.bind;
+        this.onFilterEndChange = this.onFilterEndChange.bind;
+        this.onSubmit = this.onSubmit.bind(this);
```

Comment:
```suggestion
```

The `.bind` motif is something that shouldn't be necessary in class components anymore thanks to arrow functions. I'll make some inline comments below showcasing the use of arrow functions, but they allow us to keep `this` in scope in the body of functions that are a part of a React class component

Here's a [quick article](https://medium.com/quick-code/react-quick-tip-use-class-properties-and-arrow-functions-to-avoid-binding-this-to-methods-29628aca2e25) that talks about it

Reply:
Fixed, I turned all the functions into arrow functions